service/src/upgrade.rs

```diff
@@ -86,3 +86,41 @@ pub mod fusedev_upgrade {
         Ok(())
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_failover_policy() {
+        assert_eq!(
+            FailoverPolicy::try_from("flush").unwrap(),
+            FailoverPolicy::Flush
+        );
+        assert_eq!(
+            FailoverPolicy::try_from("resend").unwrap(),
+            FailoverPolicy::Resend
+        );
+
+        let strs = vec!["flash", "Resend"];
+        for s in strs.clone().into_iter() {
+            assert!(FailoverPolicy::try_from(s).is_err());
+        }
+
+        let str = String::from("flush");
```

Comment:
Thanks for the test! Is it necessary to test String?

Reply:
Thank you for response and suggestion! I test String type transform here to increase code coverage. Indeed, testing String type is optional.